### PR TITLE
EVG-3188 Implement fmt.Stringer for APIString

### PR DESCRIPTION
**Pull request by chasinglogic**
This fixes printing errors when used in a format string such as:

```
			grip.Infof("Spawn host created with ID '%s'. Visit the hosts page in Evergreen to check on its status.", host.Id)
```

Now correctly prints:

```
evergreen-ci/evergreen master Δ ./clients/darwin_amd64/evergreen host create -d rhel70-small -k mathewrobinson
A new version is available. Run './clients/darwin_amd64/evergreen get-update --install' to download and install it.
[evergreen] 2018/04/20 12:45:00 [p=info]: Spawn host created with ID 'evg-rhel70-small-20180420164500-1674881231148019645'. Visit the hosts page in Evergreen to check on its status.
```

### CODE REVIEW

**Comment by john-m-liu** [comment]
Can you rebase? I refactor APIString recently

**Comment by chasinglogic** [comment]
@john-m-liu Done

**Comment by bsamek** [comment]
We canceled the patch build on the rebase, so I added a noop commit to trigger another build.

**Comment by bsamek** [comment]
Oh, I'm wrong, the tests actually were failing.

**Comment by bsamek** [comment]
As an alternative, you could call `model.FromAPIString` on `host.Id` in the logging statement.

File: rest/model/string.go [hunk 1]
```diff
@@ -12,3 +12,7 @@ func FromAPIString(in APIString) string {
 	}
 	return *in
 }
+
+func (as APIString) String() string {
+	return string(as)
```
Comment (john-m-liu) [inline rest/model/string.go]: This should call FromAPIString

**Comment by richardsamuels** [comment]
I don't believe this change can work at all because `type APIString *string`. 

Golang refuses to let a method be added to this type, and changing the declaration to `type APIString = *string` is a fair bit of work and introduces compatibility issues with old Go versions.

**Comment by bsamek** [comment]
> I don't believe this change can work at all because type APIString *string.

Right. I think the only thing to do is call `model.APIString` in the logging statement, not add a method on the type.

**Comment by chasinglogic** [comment]
When I originally wrote this it was `type APIString string` you're right after @john-m-liu changes making it a Stringer no longer works, I've changed this to instead call `model.FromAPIString` in the operation.

File: rest/model/string.go [hunk 2]
```diff
@@ -12,3 +12,7 @@ func FromAPIString(in APIString) string {
 	}
 	return *in
 }
+
+func (as APIString) String() string {
```
[the remark this answers is not in the document]
Comment (chasinglogic) [reply rest/model/string.go]: oversight, removed

**Pull request merged**